R/uvhydrograph-render.R

```diff
@@ -157,22 +161,100 @@ createSecondaryPlot <- function(data, month){
       plot_object <- legend(plot_object, location="below", title="", ncol=ncol, 
                                 legend_offset=legend_offset, cex=0.8, y.intersp=1.5) %>% 
         grid(nx=0, ny=NULL, equilogs=FALSE, lty=3, col="gray") %>% 
-        abline(v=secondaryInfo$plotDates, lty=3, col="gray")
+        abline(v=referenceInfo$plotDates, lty=3, col="gray")
       
-      isShift <- length(grep("shift", names(secondaryData))) > 0
+      isShift <- length(grep("shift", names(referenceData))) > 0
       if(isShift){
         plot_object <- plot_object %>% 
-          mtext(paste0(secondaryInfo$tertiary_lbl, " (", secondaryInfo$sec_units, ")"), 
+          mtext(paste0(referenceInfo$tertiary_lbl, " (", referenceInfo$ref_units, ")"), 
                               side = 4, line = 1.5) %>% 
           axis(side=4, las=0)
       }
       
       plot_object <- testCallouts(plot_object, xlimits = xlim(plot_object)$side.1)
     
-      table <- correctionsTable(secondaryData)
+      table <- correctionsTable(referenceData)
     
     } else {
-      status_msg <- paste('Corrected data missing for', data$reportMetadata$secondaryParameter)
+      status_msg <- paste('Corrected data missing for', data$reportMetadata$referenceParameter)
+    }
+  } 
+  
+  return(list(plot=plot_object, table=table, status_msg=status_msg))
+}
+
+createUpchainPlot <- function(data, month){
```

Comment:
seems like this entire function is a copy of the createReferencePlot function, other than using the word upchain rather than reference. I think that is a good indication that we should create one utils function that uses a more generic name, and I think keeping secondary plot could be fine - secondary referring to upchain or reference.